core/src/main/java/bisq/core/dao/node/parser/TxOutputParser.java

```diff
@@ -171,17 +244,118 @@ private void handleUnlockBondTx(TempTxOutput txOutput) {
+                    // Out: OpReturn
+                    // Miner fee: 1000 sat  (10 BSQ burned)
+
+
+                    // Case 2: 17 BSQ fee to burn
+                    // In: 17 BSQ
+                    // Out: burned BSQ change 7 BSQ -> BTC (7 BSQ burned)
+                    // Out: OpReturn
+                    // Miner fee: 1000 sat  (10 BSQ burned)
+                    return tempTxOutput.getIndex() >= 1;
+            }
+        }
+        return false;
+    }
+
+    private boolean isIssuanceCandidateTxOutput(TempTxOutput tempTxOutput) {
+        // If we have BSQ left as fee and we are at the second output we interpret it as a compensation request output.
+        return availableInputValue > 0 &&
+                tempTxOutput.getIndex() == 1 &&
+                optionalOpReturnType.isPresent() &&
+                (optionalOpReturnType.get() == OpReturnType.COMPENSATION_REQUEST ||
+                        optionalOpReturnType.get() == OpReturnType.REIMBURSEMENT_REQUEST);
+    }
+
+    private void handleIssuanceCandidateOutput(TempTxOutput tempTxOutput) {
+        // We do not permit more BSQ outputs after the issuance candidate.
+        prohibitMoreBsqOutputs = true;
+
+        // We store the candidate but we don't apply the TxOutputType yet as we need to verify the fee  after all
+        // outputs are parsed and check the phase. The TxParser will do that....
+        optionalIssuanceCandidate = Optional.of(tempTxOutput);
+    }
+
+    private void handleBlindVoteFeeOutput(TempTxOutput tempTxOutput) {
+        tempTxOutput.setTxOutputType(TxOutputType.BTC_OUTPUT);
+        prohibitMoreBsqOutputs = true;
+    }
+
+    private boolean isBlindVoteFeeOutput(TempTxOutput tempTxOutput) {
+        if (!optionalOpReturnType.isPresent()) {
+            return false;
+        }
+
+        if (optionalOpReturnType.get() != OpReturnType.BLIND_VOTE) {
+            return false;
+        }
+
+        // If it is the vote stake output we return false.
+        if (tempTxOutput.getIndex() == 0) {
+            return false;
+        }
+
+        // There must be a vote fee left
+        if (availableInputValue <= 0) {
+            return false;
+        }
+
+        // Burned BSQ output is last output before opReturn.
+        // We could have also a BSQ change output as last output before opReturn but that will
+        // be detected at blindVoteFee check.
+        // We always have the BSQ change before the burned BSQ output if both are present.
+        checkArgument(optionalOpReturnIndex.isPresent());
+        if (tempTxOutput.getIndex() != optionalOpReturnIndex.get() - 1) {
+            return false;
+        }
+
+        // Without checking the fee we would not be able to distinguish between 2 structurally same transactions, one
+        // where the output is burned BSQ and one where it is a BSQ change output.
+        long blindVoteFee = daoStateService.getParamValueAsCoin(Param.BLIND_VOTE_FEE, tempTxOutput.getBlockHeight()).value;
```

Comment:
I feel it would be better to disallow any other BSQ outputs than the vote stake output for blind vote txs. I think the code would be easier to read and I don't see any other use case.

Reply:
There can be an optional BSQ change output.

Inputs [1-n]: BSQ inputs for BSQ fee + stake
Inputs [0-n]: BTC inputs for miner fee
Output [1]: Mandatory BSQ output of stake
Output [0-1] Optional BSQ change output
Outputs [0-1]: BTC change output OR optional burned BSQ as BTC output
Outputs [0-1]: Output [1]: OP_RETURN with OpReturnData and amount 0
Mining fee: BTC mining fee + burned BSQ fee

Reply:
Without knowing the fee you cannot distinguish between a BSQ change and burned BSQ